### PR TITLE
Merge convert-document into ingest-file

### LINKED ISSUE #122: Consider merging convert-document into main app
At the moment, the `ingestors` will call on an HTTP service provided by `convert-document` (in this repo) to convert documents in various types (Word, Powerpoint, etc.) to PDF files, which then the ingestors know how to handle. `convert-document` the uses LibreOffice to do the conversion and returns the PDF. The reasoning behind this was:

a) We wanted to run a single instance of LibreOffice in each instance of `convert-document` and then use UNO, a built in IPC mechanism, to submit documents from the HTTP server. This would have shaved off the startup overhead for LibreOffice. However, the python UNO bindings are badly maintained and LibreOffice, when called via UNO, has a tendency to lock up when prompted to process broken office documents. After exhaustively debugging this with limited success, we've now switched `convert-document` back to a process-based model where it shells out to call `soffice --headless`. 

b) Another advantage of running `convert-document` in its own container is that we can insulate LibreOffice, a complex application that does networky things. Unlike the `ingest-file` service, `convert-document` does not need credentials for ftmstore, the task redis, or even outbound network access at all. We've been using `NetworkPolicy` in k8s production to essentially block it into its container.

Still, there are also downsides to running `c-d` in it's own container: 

a) LibreOffice can only ever run one action at the same time. So the `c-d` HTTP app implements exclusive locking, and will return HTTP 503 codes if it receives a request while already processing a document. The 503 is then used by `ingest-file` to round-robin the available `c-d` instances and to implement incremental back-off. Doing all this is quite a big of hassle and in some cases becomes a major perf bottle neck on document imports.

b) We're shipping two separate units, both with a somewhat fattish ubuntu installed. This probably also creates extra attack surface.

So if we're not going to be using the UNO API going forward, I feel like we should perhaps just inline convert-document to make the ingestors directly shell out to `soffice --headless` in their own container. It's a small security loss but a big simplification of the overall operation of the system. We also then need to plan for the memory spikes that would occur in `ingest-file` while it starts up libreoffice (ca. 200MB or so).

### PR DESCRIPTION
**Pull request by stchris**
As suggested by @pudo in #122 we are hereby giving up on `convert-document` as a standalone service which talks http to receive and convert files. Instead we are opting for shelling out (and spawning) a headless LibreOffice every time we want to convert something.

### CODE REVIEW

File: ingestors/support/convert.py [hunk 1]
```diff
@@ -15,6 +15,7 @@
 CONVERT_DIR = os.path.join(gettempdir(), "convert")
 OUT_DIR = os.path.join(CONVERT_DIR, "out")
 TIMEOUT = 3600
+CONVERT_RETRIES = 10
```
Comment (pudo) [inline ingestors/support/convert.py:18]: I'm curious if the results will ever really change after 3 or 4 tries

File: Dockerfile [hunk 2]
```diff
@@ -96,6 +96,8 @@ RUN apt-get -qq -y update \
     tesseract-ocr-aze \
     tesseract-ocr-bel \
     tesseract-ocr-uzb \
+    ### pdf convert \
+    libreoffice \
```
Comment (pudo) [inline Dockerfile:100]: Do you need to install a bunch of font packages to make this generate reasonable-looking PDFs? There was quite a lot of junk installed in the old container, most of which came about from broken PDFs being generated.

File: ingestors/support/convert.py (hunk 1 above)
Comment (stchris) [reply ingestors/support/convert.py:18]: They tend not to, indeed. We're coming down from 256 retries. Might settle on 5.

File: Dockerfile (hunk 2 above)
Comment (stchris) [reply Dockerfile:100]: Oh, I tried the minimum footprint approach, but I will need to do some more testing then. Thanks!

File: ingestors/support/convert.py (hunk 1 above)
Comment (Rosencrantz) [reply ingestors/support/convert.py:18]: I have looked and 5 or 6 seems like a nice number.

File: ingestors/misc/jsonfile.py [hunk 3]
```diff
@@ -1,6 +1,5 @@
 import json
 
-from followthemoney import model
```
Comment (Rosencrantz) [inline ingestors/misc/jsonfile.py:3]: Assuming that followthemoney was not being used in this file?

File: ingestors/support/convert.py [hunk 4]
```diff
@@ -1,21 +1,25 @@
...
 from ingestors.support.cache import CacheSupport
 from ingestors.support.temp import TempFileSupport
-from ingestors.util import explicit_resolve
 from ingestors.exc import ProcessingException
 
 log = logging.getLogger(__name__)
 
+CONVERT_DIR = os.path.join(gettempdir(), "convert")
+OUT_DIR = os.path.join(CONVERT_DIR, "out")
+TIMEOUT = 3600
```
Comment (Rosencrantz) [inline ingestors/support/convert.py:17]: Is this in milliseconds or seconds?

File: ingestors/documents/pdf.py [hunk 5]
```diff
@@ -1,7 +1,6 @@
-import logging
 
 from ingestors.ingestor import Ingestor
-from ingestors.support.pdf import PDFSupport, PdfModel
```
[the remark this answers is not in the document]
Comment (stchris) [reply ingestors/documents/pdf.py:4]: it's an unused import

File: ingestors/misc/jsonfile.py (hunk 3 above)
Comment (stchris) [reply ingestors/misc/jsonfile.py:3]: Yeah, this is an automatic linter which went too far. We're not using this here, but I shouldn't have changed this out-of-scope file.

File: ingestors/documents/pdf.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (pudo) [reply ingestors/documents/pdf.py:4]: unused by who?

File: ingestors/support/convert.py (hunk 4 above)
Comment (stchris) [reply ingestors/support/convert.py:17]: seconds, will add a comment 👍

Comment (stchris) [reply ingestors/support/convert.py:17]: and perhaps lower it to something reasonable? 600?

File: ingestors/support/convert.py [hunk 6]
```diff
@@ -35,43 +39,53 @@ def document_to_pdf(self, file_path, entity):
...
-                        stream=True,
+
+        pathlib.Path(OUT_DIR).mkdir(parents=True, exist_ok=True)
+
+        cmd = [
+            "/usr/bin/libreoffice",
+            '"-env:UserInstallation=file:///tmp"',
+            "--nologo",
+            "--headless",
+            "--nocrashreport",
```
Comment (Rosencrantz) [inline ingestors/support/convert.py:54]: Is there a world in which we'd want to try and capture this information in the future?

**Review by Rosencrantz — APPROVED** [review]
Added a few small comments.

File: ingestors/documents/pdf.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (pudo) [reply ingestors/documents/pdf.py:4]: why not? might be useful....

File: ingestors/support/convert.py (hunk 4 above)
Comment (Rosencrantz) [reply ingestors/support/convert.py:17]: Hmmm 10 minutes sounds reasonable for reasonably sized PDF documents. But what happens when the PDF is 1GB+ in size. Let's keep the TIMEOUT painfully big until we can review how quickly we can successfully process a thicc pdf.

File: ingestors/documents/pdf.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (stchris) [reply ingestors/documents/pdf.py:4]: I'm hanging up now

**Pull request merged**